include/fiction/technology/sidb_cluster_hierarchy.hpp

```diff
@@ -0,0 +1,1080 @@
+    /**
+     * The potential projection ordering.
+     */
+    pot_proj_order order{};
+    /**
+     * Default constructor, creating the empty potential projection order.
+     */
+    explicit potential_projection_order() noexcept = default;
+    /**
+     * Constructor for a potential projection from a singleton cluster onto the SiDB contained in it.
+     *
+     * @param loc_ext_pot The local external potential at the SiDB in the singleton cluster. Specifically, this is the
+     * sum of the local defect potential and the local external potential.
+     * @param base The simulation base. This defines whether positive charges are considered.
+     * @param self_projection Separates the constructor type from inter-SiDB potential projections.
+     */
+    explicit potential_projection_order(const double loc_ext_pot, const uint8_t base,
+                                        [[maybe_unused]] bool self_projection) noexcept :
+            order{base == 3 ? pot_proj_order{potential_projection{loc_ext_pot, sidb_charge_state::POSITIVE},
+                                             potential_projection{loc_ext_pot, sidb_charge_state::NEUTRAL},
+                                             potential_projection{loc_ext_pot, sidb_charge_state::NEGATIVE}} :
+                              pot_proj_order{potential_projection{loc_ext_pot, sidb_charge_state::NEUTRAL},
+                                             potential_projection{loc_ext_pot, sidb_charge_state::NEGATIVE}}}
+    {}
+    /**
+     * Constructor for a potential projection from a singleton cluster onto an SiDB.
+     *
+     * @param inter_sidb_pot The chargeless potential between the SiDB in the singleton cluster and the one projected
+     * onto, as found in the potential matrix in an associated `charge_distribution_surface` object.
+     * @param base The simulation base. This defines whether positive charges are considered.
+     */
+    explicit potential_projection_order(const double inter_sidb_pot, const uint8_t base) noexcept :
+            order{base == 3 ? pot_proj_order{potential_projection{-inter_sidb_pot, sidb_charge_state::POSITIVE},
+                                             potential_projection{0.0, sidb_charge_state::NEUTRAL},
+                                             potential_projection{inter_sidb_pot, sidb_charge_state::NEGATIVE}} :
+                              pot_proj_order{potential_projection{0.0, sidb_charge_state::NEUTRAL},
+                                             potential_projection{inter_sidb_pot, sidb_charge_state::NEGATIVE}}}
+    {}
+    /**
+     * A getter for a potential projection bound, which is the first or last item in the ordered set.
+     *
+     * @tparam bound The type of bound to obtain.
+     * @return The potential projection that forms the requested bound on the potential projection order.
+     */
+    template <bound_direction bound>
+    [[nodiscard]] constexpr inline const potential_projection& get() const noexcept
```

Comment:
can we choose a more expressive name for this function?

Reply:
`get_bound` I had earlier, but I switched it since the template param already says `bound`: e.g., 

```c++
order.get<bound_direction::UPPER>()
```

Reply:
I actually prefer ``get_bound`` 😅, it feels easier to understand when someone sees it for the first time.